### PR TITLE
theme: add citesummary view

### LINKED ISSUE #1304: Citation summary
Legacy supported a mode where several metrics where computed based on the records matching a query. The same functionality should be available on Labs, possibly implemented as an API entry-point allowing for spitting out JSON.

See: https://inspirehep.net/search?ln=en&p=author%3AB.Storaci.1&of=hcs2

### PR DESCRIPTION
**Pull request by jacquerie**
Closes #1304 
Closes #1672

### CODE REVIEW

**Comment by jacquerie** [comment]
It works (http://inspire-nightly.cern.ch/experiments/1110643/citesummary), but I probably did some atrocities with `require.js`, so I want @jmartinm's opinion on this.

File: inspirehep/modules/theme/static/js/inspire_citesummary_init.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * This file is part of INSPIRE.
+ * Copyright (C) 2014 CERN.
```
Comment (jacquerie) [inline inspirehep/modules/theme/static/js/inspire_citesummary_init.js]: Bad year.

File: inspirehep/modules/theme/static/js/settings.js [hunk 2]
```diff
@@ -132,10 +137,31 @@ require.config({
...
+    'dc-legend2': {
+      deps: [
+        'dc'
+      ]
+    },
     'holding-pen-controllers': {
       deps: [
         'angular',
-     ],
+      ],
```
Comment (jacquerie) [inline inspirehep/modules/theme/static/js/settings.js]: Sigh. To be split in its own commit.

File: inspirehep/modules/theme/static/scss/base/core.scss [hunk 3]
```diff
@@ -183,29 +182,120 @@ a[ng-click] {
...
+  -o-animation: loading-bar-spinner 400ms linear infinite;
+  animation: loading-bar-spinner 400ms linear infinite;
 }
 
-
 .modal-body {
-    max-height: calc(100vh - 212px);
-    overflow-y: auto;
+  max-height: calc(100vh - 212px);
+  overflow-y: auto;
```
Comment (jacquerie) [inline inspirehep/modules/theme/static/scss/base/core.scss]: Style changes above this line will be split in another commit.

File: inspirehep/modules/theme/static/scss/base/core.scss [hunk 4]
```diff
@@ -183,29 +182,120 @@ a[ng-click] {
...
+    }
+
+    &.journals {
+      background-color: $journals-collection;
+
+      .cs-back-link {
+        padding-bottom: 15px;
+        border-bottom: 1px solid $asbestos;
+      }
+    }
```
Comment (jacquerie) [inline inspirehep/modules/theme/static/scss/base/core.scss]: I wonder if there's a way to use mixins to DRY these.

File: inspirehep/modules/theme/templates/inspirehep_theme/citations/citation_summary.html [hunk 5]
```diff
@@ -0,0 +1,77 @@
+{#
+# This file is part of INSPIRE.
+# Copyright (C) 2015, 2016 CERN.
```
Comment (jacquerie) [inline inspirehep/modules/theme/templates/inspirehep_theme/citations/citation_summary.html]: Bad year.

File: inspirehep/modules/theme/templates/inspirehep_theme/citations/citation_summary.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
```
Comment (jacquerie) [inline inspirehep/modules/theme/templates/inspirehep_theme/citations/citation_summary.html]: Bad license.

File: inspirehep/modules/theme/views.py [hunk 7]
```diff
@@ -182,6 +182,11 @@ def data():
     )
 
 
+@blueprint.route('/<string:collection>/<int:recid>/citesummary', methods=['GET',])
+def citation_summary(collection, recid):
+    """ View for citation summary for a record """
+    return render_template('inspirehep_theme/citations/citation_summary.html', collection=collection, recid=recid)
+
```
Comment (jacquerie) [inline inspirehep/modules/theme/views.py]: Add a space here.

File: inspirehep/modules/theme/bundles.py [hunk 8]
```diff
@@ -71,12 +71,6 @@
     }
 )
 
-detailedjs = NpmBundle(
-    "js/detailed_record_init.js",
-    filters="requirejs",
-    output="gen/detailed_record.%(version)s.js",
-)
```
Comment (jacquerie) [inline inspirehep/modules/theme/bundles.py]: Also this cleanup should go in its own commit.

Comment (jmartinm) [inline inspirehep/modules/theme/bundles.py]: Why is this removed? seems to be still in `setup.py` https://github.com/inspirehep/inspire-next/pull/1699/files#diff-2eeaed663bd0d25b7e608891384b7298R239

**Comment by jmartinm** [comment]
> It works (http://inspire-nightly.cern.ch/experiments/1110643/citesummary), but I probably did some atrocities with require.js, so I want @jmartinm's opinion on this.

Nothing that calls my attention at first sight (apart from the removal of `detailedjs`)

Something specific that you did and are not sure if it is correct?

**Comment by jacquerie** [comment]
> Something specific that you did and are not sure if it is correct?

https://github.com/inspirehep/inspire-next/pull/1699/files#diff-b389934115eaa8ec351610c9ddb7496fR154

As far as I understand, `dc` does not do this exporting on its own, but just defines a `dc` variable. This configuration puts that `dc` variable in the global scope of a page that uses `citesummary.js`, which sounds pretty yucky to me.

**Comment by jmartinm** [comment]
> As far as I understand, dc does not do this exporting on its own, but just defines a dc variable. This configuration puts that dc variable in the global scope of a page that uses citesummary.js, which sounds pretty yucky to me.

I see, actually checking the source of `dc.js` it seems like it supports requirejs. There is the following code:

```javascript
if(typeof define === "function" && define.amd) {
        define(["d3", "crossfilter"], _dc);
    } else if(typeof module === "object" && module.exports) {
        var _d3 = require('d3');
        var _crossfilter = require('crossfilter');
        // When using npm + browserify, 'crossfilter' is a function,
        // since package.json specifies index.js as main function, and it
        // does special handling. When using bower + browserify,
        // there's no main in bower.json (in fact, there's no bower.json),
        // so we need to fix it.
        if (typeof _crossfilter !== "function") {
            _crossfilter = _crossfilter.crossfilter;
        }
        module.exports = _dc(_d3, _crossfilter);
    } else {
        this.dc = _dc(d3, crossfilter);
    }
```

So in theory just by adding it on `require` should work. Although I assume you tried and that wasn't the case.

**Comment by jmartinm** [comment]
I can take this branch and give it a try locally to see first hand. Will report here my findings.

**Comment by jmartinm** [comment]
To me it seems to work without the `dc` entry in the settings.js `shim` config. What was the error you were seeing?

I only get every now and then an error when brushing:

```
Error: <g> attribute transform: Trailing garbage, "translate(0,NaN)"
```

File: inspirehep/modules/theme/bundles.py (hunk 8 above)
Comment (jmartinm) [inline inspirehep/modules/theme/bundles.py]: Oh I see 👍

**Comment by jacquerie** [comment]
> Although I assume you tried and that wasn't the case.

Yup. Even on Nightly. What configuration are you using? `DEBUG`, `ASSETS_DEBUG`...?

**Comment by jmartinm** [comment]
> Yup. Even on Nightly. What configuration are you using? DEBUG, ASSETS_DEBUG...?

I was using all production style. No `DEBUG` or `ASSETS_DEBUG`.

**Comment by jacquerie** [comment]
Another bug I spotted: the `citesummary`view should handle gracefully the case in which a paper has no citations. For example: http://inspire-nightly.cern.ch/api/literature/1483042/citesummary results in http://inspire-nightly.cern.ch/literature/1483042/citesummary (check the error in the console). This can happen in real data (it's not an artifact of a non-full migration): https://inspirehep.net/record/1483042

**Comment by jacquerie** [comment]
Another bug I spotted: http://inspire-nightly.cern.ch/literature/451647/citesummary (again, check the console).

**Comment by kaplun** [comment]
Before we deploy this we need to solve performance issues.

**Comment by jacquerie** [comment]
Postponed, will be resurrected if necessary.
